src/python/WMCore/Services/RucioConMon/RucioConMon.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+# coding=utf-8
+"""
+Rucio Consistency Monitor Service class developed on top of the native WMCore
+Service APIs providing custom output and handling as necessary for the
+CMS Workload Management system
+"""
+
+from __future__ import division, print_function, absolute_import
+
+from urllib.parse import urlencode
+
+import json
+import logging
+
+from WMCore.Services.Service import Service
+
+
+class RucioConMon(Service):
+    """
+    API for dealing with retrieving information from Rucio Consistency Monitor
+    """
+
+    def __init__(self, url, logger=None, configDict=None):
+        """
+        Init method for the RucioConMon Class
+        """
+        configDict = configDict or {}
+        configDict.setdefault('endpoint', url)
+        configDict.setdefault('cacheduration', 1)  # in hours
+        configDict.setdefault('accept_type', 'application/json')
+        configDict.setdefault('content_type', 'application/json')
+        configDict['logger'] = logger if logger else logging.getLogger()
+        super(RucioConMon, self).__init__(configDict)
```

Comment:
With this super `__init__` at this position, you can be potentially overriding all the defaults and configurations set above. Can you please move it to one of the first lines in this method, and only then define this class extra arguments?

Reply:
This is how it is defined in CRIC Service [1]. So if it is wrong her .. well then it is wrong there too.
[1]
https://github.com/dmwm/WMCore/blame/eeb8ab2c7083d95a1dc50879d48287f508b33709/src/python/WMCore/Services/CRIC/CRIC.py#L35-L47

Reply:
That's what happens when you review something on Saturday 1am... I thought the setdefault was on the self object, not on configDict.
You're correct, Todor. It's good like this.